docker-compose-dev.yml

```diff
@@ -7,19 +7,30 @@ services:
     build:
       context: ./
       dockerfile: Dockerfile.dev
-    # Not sure why ports aren't working. Using host as a hack
-    # ports:
-    #   - "9999:9999"
-    network_mode: "host"
+    ports:
+      - "9999:9999"
     environment:
-      - GOTRUE_DB_MIGRATIONS_PATH=/go/src/github.com/netlify/gotrue/migrations
-    volumes:
-      - ./:/go/src/github.com/netlify/gotrue
-    command: CompileDaemon --build="make build" --directory=/go/src/github.com/netlify/gotrue --recursive=true -pattern=(.+\.go|.+\.env) -exclude=gotrue -exclude=gotrue-arm64 --command=/go/src/github.com/netlify/gotrue/gotrue
+      # Update the docker environment accordingly based on example.env
+      GOTRUE_SITE_URL: ${SITE_URL}
```

Comment:
I'm surprised you need these :thinking: docker-compose should be pulling them automatically from a created `.env` file. If you're wanting to use values in `example.env` by default then maybe consider the `--env-file` arg in the makefile so the values only exist in one place: https://docs.docker.com/compose/environment-variables/#the-env-file

Reply:
yeah maybe it would be better if we have a separate env file for docker (`.env.docker`) because the `API_HOST=0.0.0.0` on docker while `API_HOST=localhost` if running locally without docker

Reply:
Setting the `API_HOST` value is actually a mac specific issue. I have a mac for work and we have to do the same for our docker containers. On Linux the container can be accessed on the host using `localhost` and a separate `.env.` wouldn't be needed. This gets back to my original issue with the port forwarding not working on linux. I have other projects with the same setup (Go server inside docker) and setting the ports has no problem and lets me use `localhost` to access the server.